source/opt/local_access_chain_convert_pass.cpp

```diff
@@ -247,9 +262,11 @@ Pass::Status LocalAccessChainConvertPass::ConvertLocalAccessChains(
     for (auto ii = bi->begin(); ii != bi->end(); ++ii) {
       switch (ii->opcode()) {
         case SpvOpLoad: {
+          if (HasOpenCL100DebugRef(&*ii)) break;
           uint32_t varId;
           Instruction* ptrInst = GetPtr(&*ii, &varId);
           if (!IsNonPtrAccessChain(ptrInst->opcode())) break;
+          if (HasOpenCL100DebugRef(ptrInst)) break;
```

Comment:
Why should we not change the access chain?  I believe this pass leaves the original pointer around to be cleaned up by dead code elimination (I could be wrong about that).  If it does, then why does replacing the access chain make the debug information incorrect?

Same for the store below.

Reply:
I agree. I thought it will be killed by this pass, but it will not.
I simply remove this if-statement.